javasource/restservices/util/JsonDeserializer.java

```diff
@@ -66,101 +66,180 @@ else if (jsonValue instanceof Long || jsonValue instanceof Double || jsonValue i
+					String otherSideType = target.getMetaObject().getMetaAssociationParent(targetattr).getChild().getName();
+					if(otherSideType.equals(target.getMetaObject().getName()))
+						otherSideType= target.getMetaObject().getMetaAssociationChild(targetattr).getParent().getName();
+
+					if (!object.isNull(attr))
+						((MendixObjectReference)member).setValue(context, readJsonDataIntoMendixObject(context, object.get(attr), otherSideType, autoResolve));
+				}
+
+				//ReferenceSet
+				else if (member instanceof MendixObjectReferenceSet){
+					String otherSideType = target.getMetaObject().getMetaAssociationParent(targetattr).getChild().getName();
+					if(otherSideType.equals(target.getMetaObject().getName()))
+						otherSideType= target.getMetaObject().getMetaAssociationChild(targetattr).getParent().getName();
+
+					JSONArray children = object.getJSONArray(attr);
+					List<IMendixIdentifier> ids = new ArrayList<IMendixIdentifier>();
+
+					for(int i = 0; i < children.length(); i++) {
+						IMendixIdentifier child = readJsonDataIntoMendixObject(context, children.get(i), otherSideType, autoResolve);
+						if (child != null) {
+							/*
+							 * The core.createMendixIdentifier should be unnecessary, however, there is a bug there, see
+							 * support ticket 102188
+							 */
+							ids.add(Core.createMendixIdentifier(child.toLong()));
+						}
 					}
+
+					((MendixObjectReferenceSet)member).setValue(context, ids);
 				}
-				
-				((MendixObjectReferenceSet)member).setValue(context, ids);
+
+				//Primitive member
+				else
+				{
+					IMetaPrimitive primitive = target.getMetaObject().getMetaPrimitive(targetattr);
+					if (primitive.getType() != PrimitiveType.AutoNumber)
+						target.setValue(context, targetattr, jsonAttributeToPrimitive(primitive, object, attr));
+				}
+
 			}
-			
-			//Primitive member
-			else if (target.hasMember(targetattr)){
-				IMetaPrimitive primitive = target.getMetaObject().getMetaPrimitive(targetattr);
-				if (primitive.getType() != PrimitiveType.AutoNumber)
-					target.setValue(context, targetattr, jsonAttributeToPrimitive(primitive, object, attr));
+			else
+			{
+				IMetaAssociation association =  Core.getMetaAssociation(targetattr);
+				String otherSideType = association.getParent().getName();
+				IMendixObject otherSideTarget = Core.instantiate(context, otherSideType);
+				Map<String, String> otherAttributeNameMap = buildAttributeNameMap(otherSideTarget.getMetaObject());
+				String otherTargetattr =  otherAttributeNameMap.get(attr.toLowerCase().replaceAll("[^a-zA-Z0-9_]","_"));
+
+				IMendixObjectMember<?> otherMember = otherSideTarget.getMember(context, otherTargetattr);
+
+				//Reference for arrays
+				if (otherMember instanceof MendixObjectReference)
+				{
+					JSONArray jsonArray = object.getJSONArray(attr);
+					readJsonArrayIntoMendixObject(context, jsonArray, otherSideTarget, autoResolve, association.getName().split("\\.")[1], target.getId());
+				}

```

Comment:
I think this needs some comments as the code is a bit counterintuitive, the check is on a `MendixObjectReference` and after that stuff is threted as an array. I guess we are dealing here with an inversion relation `parent <---assoc ---- child`

What happens if the the assoc is actually a `MendixObjectReferenceSet`? Or if the `child` json data is not an array but a single object?